src/+otp/+kortewegdevries/KortewegdeVriesProblem.m

```diff
@@ -0,0 +1,102 @@
+classdef KortewegdeVriesProblem < otp.Problem
+
+    % Ascher, Uri M. and Robert I. McLachlan. “On Symplectic and 
+    % Multisymplectic Schemes for the KdV Equation.” 
+    % Journal of Scientific Computing 25 (2005): 83-104.
+    %
+    % Robert M. Miura, Clifford S. Gardner, Martin D. Kruskal;
+    % Korteweg‐de Vries Equation and Generalizations. II. 
+    % Existence of Conservation Laws and Constants of Motion. 
+    % J. Math. Phys. 1 August 1968; 9 (8): 1204–1209. 
+    % https://doi.org/10.1063/1.1664701
+
+    methods
+        function obj = KortewegdeVriesProblem(timeSpan, y0, parameters)
+            
+            obj@otp.Problem('Korteweg de Vries Equation', [], ...
+                timeSpan, y0, parameters);
+            
+        end
+    end
+    
+    properties (SetAccess = private)
+        DistanceFunction
+        numInvariants
+        Invariant
+        JacobianInvariant
+    end
+    
+    methods (Static)
+      
+    end
+    
+    methods (Access = protected)
+        function onSettingsChanged(obj)
+                        
+            Xdomain = obj.Parameters.Domain;
+            Nx = obj.Parameters.Nx;
+            Theta = obj.Parameters.Theta;
+            Alpha = obj.Parameters.Alpha;
+            Nu = obj.Parameters.Nu;
+            Rho = obj.Parameters.Rho;
+
+            if obj.NumVars ~= Nx
+                warning('OTP:inconsistentNumVars', ...
+                    'NumVars is %d, but there are %d grid points', ...
+                    obj.NumVars, Nx);
+            end
+            
+            bc = 'C';
+            
+            Dx = otp.utils.pde.Dd(Nx, Xdomain, 1, 1, bc);
+            DxT = Dx.';
+            
+            dx = obj.Parameters.Grid(2) - obj.Parameters.Grid(1);
+
+            % NO support for 3rd derivative central difference in utils
+
+            xx = sparse(1, Nx);
+            xx(2) = -1;
+            xx(3) = 1/2;
+            xx(Nx) = 1;
+            xx(Nx-1) = -1/2;
+            xx = xx/(dx*dx*dx);
+            D3x = toeplitz([xx(1) fliplr(xx(2:end))], xx);
```

Comment:
This can be done using spdiags. Please use that

Reply:
This is straightforward and simpler. all you need to do is define one row and MATLAB deals with the rest

Reply:
I'll let steven decide. does this at least produce a sparse matrix?

Reply:
Yeah

Reply:
made this more compact. 

xx = sparse(1, Nx);
xx([2, 3, Nx-1, Nx]) = ([-1, 1/2, -1/2, 1])/(dx*dx*dx);
D3x = toeplitz([xx(1) fliplr(xx(2:end))], xx);